### ISSUE TITLE
`unused_braces` false positive with assignment to wildcard

### ISSUE THREAD

**Issue by kadiwa4** [rust-lang/rust#116536]
This code:
```rust
fn main() {
    let e = std::iter::empty::<()>();
    _ = { e };
    // for _ in e {} // uncomment to verify that e was dropped
}
```

Gives this warning:
```
warning: unnecessary braces around assigned value
 --> src/main.rs:3:9
  |
3 |     _ = { e };
  |         ^^ ^^
  |
  = note: `#[warn(unused_braces)]` on by default
help: remove these braces
  |
3 -     _ = { e };
3 +     _ = e;
  |
```

However, the braces are not unnecessary. `_ = { e };` does the same thing as `drop(e);`, but `_ = e;` doesn't do anything (not even move `e`).
You also can't just write `{ e };`, that would trigger a `unused_must_use` warning.
I think the lint shouldn't trigger here.

There are also more complicated examples like this:

```rust
use std::fmt::Error;

fn main() {
    let e = (std::iter::empty::<()>(), Error);
    (_, Error) = { e };
    // for _ in e.0 {} // uncomment to verify that e was dropped
}
```

Where it isn't even possible to tell if the braces are unnecessary just based on the AST. But surely no-one writes code like that :)

## Version
`rustc 1.75.0-nightly (2023-10-07)`

@rustbot label: T-compiler A-lint

**Comment by mj10021** [comment]
fwiw changing `_ = ` to `let _ = ` gets rid of the lint
